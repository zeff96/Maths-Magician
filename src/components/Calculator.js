import './Calculator.scss';
import { useState } from 'react';
import calculate from '../Logic/Calculate';

const Calculator = () => {
  const [total, setTotal] = useState('');
  const [next, setNext] = useState('');
  const [operation, setOperation] = useState('');

  const handleClick = (e) => {
    const results = calculate({ total, next, operation }, e.target.textContent);
    setTotal(results.total);
    setNext(results.next);
    setOperation(results.operation);
  };
  const Digits = () => {
    const nums = [];
    for (let i = 9; i >= 1; i -= 1) {
      nums.push(
        <button type="button" key={i.toString()} onClick={handleClick}>
          {i}
        </button>,
      );
    }
    return nums;
  };

  return (
    <div className="wrapper">
      <h2>Let&#39;s do some maths!</h2>
      <div className="calculator">
        <div className="display" data-testid="display">
          {total}
          {operation}
          {next}
        </div>
        <div className="operators">
          <div className="top">
            <button type="button" onClick={handleClick}>
              AC
            </button>
            <button type="button" onClick={handleClick}>
              +/-
            </button>
            <button type="button" onClick={handleClick}>
              %
            </button>
          </div>
          <div className="side">
            <button type="button" onClick={handleClick}>
              รท
            </button>
            <button type="button" onClick={handleClick}>
              x
            </button>
            <button type="button" onClick={handleClick}>
              -
            </button>
            <button type="button" onClick={handleClick}>
              +
            </button>
            <button data-testid="result" type="button" onClick={handleClick}>
              =
            </button>
          </div>
          <div className="digits">
            <Digits />
            <button className="zero" type="button" onClick={handleClick}>
              0
            </button>
            <button type="button" onClick={handleClick}>
              .
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
