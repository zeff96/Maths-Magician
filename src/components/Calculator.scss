$primary-color: #fff;
$secondary-color: #9b9898;
$tertiary-color: #ffa500;
$font-family: helvetica, san-serif;

* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: $font-family;
}

.calculator {
  display: grid;
  width: 30rem;
  box-shadow: 0 3px 8px rgba(0, 0, 0, 0.5);

  .display {
    background-color: $secondary-color;
    text-align: right;
    padding: 1.5rem 0.2rem;
    color: $primary-color;
    font-size: 1.6rem;
  }

  .operators {
    display: grid;

    button {
      cursor: pointer;
      appearance: none;
      outline: none;
      padding: 1.5rem;
      text-align: center;
      border: 1px solid rgb(179, 176, 176);
    }

    button:hover {
      opacity: 0.5;
    }

    .top {
      grid-column: 1/3;
      grid-row: 1/1;
      display: grid;
      grid-template-columns: repeat(3, 1fr);

      button {
        background-color: $primary-color;
      }
    }

    .side {
      grid-column: 3/4;
      grid-row: 1/4;

      button {
        display: block;
        width: 100%;
        background-color: $tertiary-color;
      }
    }

    .digits {
      grid-column: 1/3;
      display: grid;
      grid-template-columns: repeat(3, 1fr);

      button {
        background-color: $primary-color;
      }

      .zero {
        grid-column: 1/3;
      }
    }
  }
}
