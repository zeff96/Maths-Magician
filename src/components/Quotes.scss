.quotes {
  width: 50rem;
  margin-bottom: 3rem;
  font-family: Helvetica, sans-serif;
  font-size: 1.6rem;
}
