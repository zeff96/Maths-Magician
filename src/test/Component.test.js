import { render, screen } from '@testing-library/react';
import { Route, MemoryRouter } from 'react-router-dom';
import Calculator from '../components/Calculator';
import HomePage from '../components/Home';
import Navlinks from '../components/Navbar';
import Quotes from '../components/Quote';

describe('component test', () => {
  it('render home title', () => {
    render(<HomePage />);
    const header = screen.getByText(/Welcome to our page!/i);
    expect(header).toMatchSnapshot();
  });

  it('render calculator title', () => {
    render(<Calculator />);
    const header = screen.getByText(/Let's do some maths!/i);
    expect(header).toMatchSnapshot();
  });

  it('render Navlink title', () => {
    <Route>
      render(
      <Navlinks />
      ); const header = screen.getByText(/Maths Magician/i);
      expect(header).toMatchSnapshot();
    </Route>;
  });
  it('simulate user click event', () => {
    <MemoryRouter>
      render(
      <Navlinks />
      ) const linkEl = screen.getByText(/home/i);
      fireEvent.click(linkEl); expect(getByText(/Welcome to our
      page!/i)).toBeInTheDocument();
    </MemoryRouter>;
  });

  it('renders a div element with class name "quotes"', async () => {
    render(<Quotes />);
    const divElement = await screen.getByTestId('quotes');

    expect(divElement).toMatchSnapshot();
  });
});
