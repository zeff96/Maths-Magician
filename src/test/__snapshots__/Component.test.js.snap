// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component test render calculator title 1`] = `
<h2>
  Let's do some maths!
</h2>
`;

exports[`component test render home title 1`] = `
<h2
  class="home-title"
>
  Welcome to our page!
</h2>
`;

exports[`component test renders a div element with class name "quotes" 1`] = `
<div
  class="quotes"
  data-testid="quotes"
>
  <div>
    Loading...
  </div>
</div>
`;
